lib/chef/provisioning/fog_driver/driver.rb

```diff
@@ -424,46 +426,82 @@ def wait_for_transport(action_handler, machine_spec, machine_options, server)
       end
     end
 
-    def attach_floating_ips(action_handler, machine_spec, machine_options, server)
-      # TODO this is not particularly idempotent. OK, it is not idempotent AT ALL.  Fix.
-      if option_for(machine_options, :floating_ip_pool)
-        Chef::Log.info 'Attaching IP from pool'
-        action_handler.perform_action "attach floating IP from #{option_for(machine_options, :floating_ip_pool)} pool" do
-          attach_ip_from_pool(server, option_for(machine_options, :floating_ip_pool))
+    def converge_floating_ips(action_handler, machine_spec, machine_options, server)
```

Comment:
This is nice and concise, as a function, without being lossfully compact.  I like :)